File: test/fixtures/reference/guard-expressions.expected.js
(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!true) throw new Error("guard failed");
          return 42;
        })() == 42
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!(5 > 3)) throw new Error("guard failed");
          return 100;
        })() == 100
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!(10 > 0)) throw new Error("guard 'positive' failed");
          return 10;
        })() == 10
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!(7 > 0)) throw new Error("value must be positive");
          return 7;
        })() == 7
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!(10 > 0)) throw new Error("guard failed");
          if (!(10 < 100)) throw new Error("guard failed");
          return 10;
        })() == 10
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!(5 > 0)) throw new Error("guard 'positive' failed");
          if (!(5 < 10)) throw new Error("guard 'small' failed");
          return 5;
        })() == 5
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          const x = 5;
          return (() => {
            if (!(x > 0)) throw new Error("guard failed");
            return x * 2;
          })();
        })() == 10
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!true) throw new Error("check failed");
          return 99;
        })() == 99
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!(3 > 0)) throw new Error("check 'valid' failed");
          return 3;
        })() == 3
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          const x = 10;
          return (() => {
            if (!(x > 0)) throw new Error("check failed");
            return x + 5;
          })();
        })() == 15
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!(1 > 0)) throw new Error("guard failed");
          return (() => {
            const x = 10;
            const y = 20;
            return x + y;
          })();
        })() == 30
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!true) throw new Error("guard failed");
          return (() => {
            if (!true) throw new Error("guard failed");
            return 7;
          })();
        })() == 7
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

(function (_) {
  return (function () {
    if (
      !(
        (() => {
          if (!true) throw new Error("guard failed");
          return true ? 1 : 2;
        })() == 1
      )
    )
      throw new Error("Assertion failed");
    return true;
  })();
})(null);

File: test/fixtures/reference/guard-expressions.expected.py

(lambda _: kAssert((kAssert(True, "guard failed"), 42)[-1] == 42))(None)


(lambda _: kAssert((kAssert(5 > 3, "guard failed"), 100)[-1] == 100))(None)


(lambda _: kAssert((kAssert(10 > 0, "guard 'positive' failed"), 10)[-1] == 10))(None)


(lambda _: kAssert((kAssert(7 > 0, "value must be positive"), 7)[-1] == 7))(None)


(lambda _: kAssert((kAssert(10 > 0, "guard failed"), kAssert(10 < 100, "guard failed"), 10)[-1] == 10))(None)


(lambda _: kAssert((kAssert(5 > 0, "guard 'positive' failed"), kAssert(5 < 10, "guard 'small' failed"), 5)[-1] == 5))(None)


(lambda _: kAssert((x := 5, (kAssert(x > 0, "guard failed"), x * 2)[-1])[-1] == 10))(None)


(lambda _: kAssert((kAssert(True, "check failed"), 99)[-1] == 99))(None)


(lambda _: kAssert((kAssert(3 > 0, "check 'valid' failed"), 3)[-1] == 3))(None)


(lambda _: kAssert((x := 10, (kAssert(x > 0, "check failed"), x + 5)[-1])[-1] == 15))(None)


(lambda _: kAssert((kAssert(1 > 0, "guard failed"), (x := 10, y := 20, x + y)[-1])[-1] == 30))(None)


(lambda _: kAssert((kAssert(True, "guard failed"), (kAssert(True, "guard failed"), 7)[-1])[-1] == 7))(None)


(lambda _: kAssert((kAssert(True, "guard failed"), (1) if (True) else (2))[-1] == 1))(None)

